src/Raven.Server/Documents/DocumentDatabase.cs

```diff
@@ -458,11 +458,14 @@ private async Task ExecuteClusterTransactionTask()
                 var topology = ServerStore.LoadDatabaseTopology(Name);
                 if (topology.Promotables.Contains(ServerStore.NodeTag))
                 {
-                    await Task.Delay(1000);
+                    await Task.Delay(1000, DatabaseShutdown);
+                    if (DatabaseShutdown.IsCancellationRequested)
```

Comment:
Whats the point of checking it here since we're passing `DatabaseShutdown` one line above?

Reply:
remvoed